src/utils/path.js

```diff
@@ -0,0 +1,102 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import type { CallNodePath } from '../types/profile-derived';
+
+export function arePathsEqual(a: CallNodePath, b: CallNodePath): boolean {
+  if (a === b) {
```

Comment:
We should document this type of trick in our end-user docs on how to write fast code. Sort transitive work by cheap/easy to hard/slow.

Reply:
Yep why not.

I also like the "fail faster" trick. Usually we want algorithms to fail faster, that's where we wins precious time. I learnt that especially when working with regexps :-D

In this specific case, I'm not entirely sure that iterating from the end will make it fail faster. It's more an educated guess: the idea that most callnodes share a common prefix with a lot of other callnodes.

But (for example) for inverted trees likely the opposite happens. So maybe we could have a parameter to this function to decide the direction, if we ever see it happening in profile.